src/components/Board.tsx

```diff
@@ -0,0 +1,90 @@
+import { useEffect, useState, useCallback } from "react";
+import { Square } from "./Square";
+import "../css/board.css";
+
+interface SquareInterface {
+    pieceColor: string,
+    color: string,
+    piece: string
+}
+
+export function Board() {
+    let [squares, setSquares] = useState<Array<Array<SquareInterface>>>();
+    let [firstRowOrder] = useState([

```

Comment:
Maybe `backRankPieceOrder` would be a better name?

Also, this doesn't need to be set in state. this can be a const outside of the component:

```
const BACK_RANK_PIECE_ORDER = [
  ChessPieces.Rook,
  . . .
]
```